### ISSUE TITLE
Grabber not working for bits.blogs.nytimes

### ISSUE THREAD

**Issue by aschilling** [fguillot/picoFeed#227]
Hi everybody,

I cannot fetch the content of the nytimes blog using the following rule:

    <?php
    return array(
        'grabber' => array(
            '%.*%' => array(
                'test_url' => 'http://bits.blogs.nytimes.com/2015/12/11/a-protest-app-rises-to-the-top-in-taiwan/',
                'body' => array(
                    '//div[@class="entry-content"]',
                ),
                'strip' => array(
                ),
            )
        )
    );

Sorry if I made a basic mistake but I am struggling with this for quite a bit now.

**Comment by mkresin** [comment]
I see what's your problem.

Please test your rulefile using the picofeed commandline client:

```
$ php ./picofeed grabber "http://bits.blogs.nytimes.com/2015/12/11/a-protest-app-rises-to-the-top-in-taiwan/"
```

And you will notice that the wrong rulefile is picked up:

```
[2015-12-17 13:19:40] PicoFeed\Scraper\RuleLoader Load rule: .nytimes.com
```

Long story short: The .nytimes.com.php is picked up first and the .nytimes.com part of the filename, indicates that this rulefile is responsible for any subdomain of nytimes.com.

I see two options to solve your issue:

- add your filter expression to .nytimes.com.php
- rename .nytimes.com.php to www.nytimes.com.php and create a blog.nytimes.com.php file with your filter rules

**Comment by aschilling** [comment]
Thank you very much for your reply. Indeed, I made this mistake.

In the aftermath of debugging the error I noticed that there are two files in the rule directory (.nytimes.com.php and .blogs.nytimes.com.php) should this not also be merged?

However, even if I modify the .nytimes rule correspondigly, it does not work. In particular a "REFUSE_COOKIE_ERROR=SHOW_ERROR" is put behind each url.

Can you reproduce this misbehavior?

**Comment by mkresin** [comment]
> In the aftermath of debugging the error I noticed that there are two files in the rule directory (.nytimes.com.php and .blogs.nytimes.com.php) should this not also be merged?

Good catch! It's interessting that both rule files where added with https://github.com/fguillot/picoFeed/commit/99b1e729661eb2481472f02714df42d957c4f8c3 a long time ago. I suspect it worked once and got broken in the meantime.

Anyway. Please send a PR with:

- .nytimes.com.php renamed to www.nytimes.com.php
- the leading dot removed from all rule files
- adjust blogs.nytimes.com.php to your needs

> However, even if I modify the .nytimes rule correspondigly, it does not work. In particular a "REFUSE_COOKIE_ERROR=SHOW_ERROR" is put behind each url.

From my logfile:

```
[243] => [2015-12-18 14:05:16] PicoFeed\Client\Curl cURL effective url: https://myaccount.nytimes.com/auth/login?URI=http%3A%2F%2Fbits.blogs.nytimes.com%2F2012%2F01%2F16%2Fwikipedia-plans-to-go-dark-on-wednesday-to-protest-sopa%2F%3F_r%3D5&REFUSE_COOKIE_ERROR=SHOW_ERROR
```

Loks like a [paywall](https://en.wikipedia.org/wiki/Paywall) to me.

**Comment by aschilling** [comment]
Thanks for your help. Unfortunately, nytimes somehow redirects all blog posts to this paywall, however, these articles are publicly accessible and liked to in the feed. Have a look at:

    "http://bits.blogs.nytimes.com/2015/12/11/a-protest-app-rises-to-the-top-in-taiwan/"
